### ISSUE TITLE
SharedPreferences crash on startup application created with flutter 2.8

### ISSUE THREAD

**Issue by Silvere1** [flutter/flutter#95031]
## SharedPreferences crash on startup Android application created with flutter 2.8

1. when i call the instance in a simple function to perform the operations, it works without problem.
But to retrieve user preferences before starting the application, it does not work. I am seeing this problem with the new projects that I create with flutter 2.8 
2. I even tried the latest version of SharedPrefences but nothing. No satisfaction.
But in the projects I created with flutter 2.5 everything works fine when I call `SharedPreferences.getInstance()` before `runApp();`

**Actual results: App no run !** 

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:shared_preferences/shared_preferences.dart';

void main() async {
  SharedPreferences preferences = await SharedPreferences.getInstance();
  preferences.setBool("my_bool", false);
  runApp(const MyApp());
}
```

</details>

<details>
  <summary>Logs</summary>

```
E/flutter (19945): [ERROR:flutter/lib/ui/ui_dart_state.cc(209)] Unhandled Exception: Null check operator used on a null value
E/flutter (19945): #0      MethodChannel.binaryMessenger (package:flutter/src/services/platform_channel.dart:132:86)
E/flutter (19945): #1      MethodChannel._invokeMethod (package:flutter/src/services/platform_channel.dart:157:36)
E/flutter (19945): #2      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:340:12)
E/flutter (19945): #3      MethodChannel.invokeMapMethod (package:flutter/src/services/platform_channel.dart:367:49)
E/flutter (19945): #4      MethodChannelSharedPreferencesStore.getAll (package:shared_preferences_platform_interface/method_channel_shared_preferences.dart:44:25)
E/flutter (19945): #5      SharedPreferences._getSharedPreferencesMap (package:shared_preferences/shared_preferences.dart:164:57)
E/flutter (19945): #6      SharedPreferences.getInstance (package:shared_preferences/shared_preferences.dart:33:19)
```

```
```

```
```

</details>

**Comment by stuartmorgan** [comment]
This has never been supported without calling `WidgetsFlutterBinding.ensureInitialized();`; if it worked in your app in 2.5 that was entirely luck.

Closing as a duplicate of https://github.com/flutter/flutter/issues/78028

**Comment by Silvere1** [comment]
Thank you very much. It was a stupid mistake on my part. 😅I didn't even pay attention to that level
